src/build/env.py

```diff
@@ -83,13 +83,29 @@ def _has_valid_pip(purelib: str) -> bool:
     else:
         from importlib import metadata
 
-    pip_distribution = next(iter(metadata.distributions(name='pip', path=[purelib])))
+    pip_distribution = next(iter(metadata.distributions(name='pip', **distargs)))
 
     current_pip_version = packaging.version.Version(pip_distribution.version)
 
     return current_pip_version >= packaging.version.Version(_minimum_pip_version())
 
 
+@functools.lru_cache(maxsize=None)
+def _valid_global_pip() -> bool | None:
+    """
+    This checks for a valid global pip. Returns None if the prerequisites are
+    not available (Python 3.7 only) or pip is missing, False if Pip is too old,
+    and True if it can be used.
+    """
+
+    try:
+        return _has_valid_pip()
+    except ModuleNotFoundError:  # Python 3.7 only
```

Comment:
Why does this happen?

Reply:
I would prefer if we always threw this from `_has_valid_pip` instead of the cryptic `StopIteration` error.

Reply:
Okay. And the error was if importlib_metadata was missing, though we have it as a dep, so that would only be for bootstrapping. This makes it have the same signature as pip missing, though, so no need to special case.